packages/pluggableWidgets/line-chart-native/src/utils/SeriesLoaderHook.ts

```diff
@@ -0,0 +1,226 @@
+                return group.groupByAttributeValue.eq(groupByAttributeValue.value);
+            }
+            return group.groupByAttributeValue === groupByAttributeValue.value;
+        });
+
+        if (group) {
+            group.items.push(item);
+        } else {
+            const newDataSourceItemGroup: DataSourceItemGroup = {
+                groupByAttributeValue: groupByAttributeValue.value,
+                items: [item]
+            };
+
+            if (dynamicSeriesName) {
+                const dynamicSeriesNameValue = dynamicSeriesName(item);
+
+                if (dynamicSeriesNameValue.value === undefined) {
+                    return null;
+                }
+
+                newDataSourceItemGroup.dynamicSeriesNameValue = dynamicSeriesNameValue.value;
+            }
+
+            if (dynamicStylePropertyName) {
+                const dynamicStylePropertyNameValue = dynamicStylePropertyName(item);
+
+                if (dynamicStylePropertyNameValue.value === undefined) {
+                    return null;
+                }
+
+                newDataSourceItemGroup.dynamicStylePropertyNameValue = dynamicStylePropertyNameValue.value;
+            }
+
+            dataSourceItemGroupsResult.push(newDataSourceItemGroup);
+        }
+    }
+
+    return dataSourceItemGroupsResult;
+}
+
+function extractDataPoints(series: SeriesType, dataSourceItems?: ObjectItem[]): DataPointsExtraction | null {
+    const xValue = series.type === "static" ? ensure(series.staticXValue) : ensure(series.dynamicXValue);
+    const yValue = series.type === "static" ? ensure(series.staticYValue) : ensure(series.dynamicYValue);
+
+    if (!dataSourceItems) {
+        const dataSource =
+            series.type === "static" ? ensure(series.staticDataSource) : ensure(series.dynamicDataSource);
+
+        if (!dataSource.items) {
+            return null;
+        }
+
+        dataSourceItems = dataSource.items;
+    }
+
+    const dataPointsExtraction: DataPointsExtraction = { dataPoints: [] };
+
+    for (const item of dataSourceItems) {
+        const x = xValue(item);
+        const y = yValue(item);
+
+        if (!x.value || !y.value) {
+            return null;
+        }
+
+        if (!dataPointsExtraction.xFormatter && !dataPointsExtraction.yFormatter) {
+            dataPointsExtraction.xFormatter = (xValue: number | Date) => {
+                return x.formatter.format(typeof xValue === "number" ? new Big(xValue) : xValue);
+            };
+            dataPointsExtraction.yFormatter = (yValue: number | Date) =>
+                y.formatter.format(typeof yValue === "number" ? new Big(yValue) : yValue);
+        }
+
+        dataPointsExtraction.dataPoints.push({
+            x: x.value instanceof Date ? x.value : (Number(x.value.toString()) as any), // Cast as any because data type will never differ for data points within a series
```

Comment:
🎄

Reply:
How would you like to see this improved?

Reply:
I liked! 😛

Reply:
Ow haha, thanks!